### DISCUSSION TITLE
complete trash support

### DISCUSSION THREAD

**Discussion by ciros88** [doublecmd/doublecmd#1108] [Suggestions]
### Description

please consider to add a button for showing the trash bin content, providing the following features:

- display files with their original path
- provide empty trash functionality
- provide files restoration functionality (thru a dedicated entry on right mouse button context menu)
- display when trash is empty or full

### Proposed technical implementation details (optional)

the trash bin button opens trash:/// path

**Comment by alexx2000** [comment]
DC 1.1.0 already has a trash "button".

**Comment by antodc** [comment]
Great, but... where is it?

**Comment by antodc** [comment]
Apologize for the previous comment. I am still using 1.0.11...
Thanks a lot for this addition in 1.1.0!!

**Comment by ciros88** [comment]
oh sorry, i saw 1.0.11 as last release, and the 1.1.0 alpha changelog doesn't seem to list this feature.
i've tried 1.1.0 alpha and i didn't found the trash button O.o where it is supposed to be? anyway after going manually to trash:/// (i have my ctrl+d shortuct) now i can see the 'restore' context menu entry.  also i didn't found the empty trash functionality neither the original path for all displayed files

**Comment by alexx2000** [comment]
It is in the VFS list (see [1](https://doublecmd.github.io/doc/en/help.html#cm_OpenVirtualFileSystemList), [2](https://doublecmd.github.io/doc/en/help.html#iface_drive_pane)). If you open trash via "Trash" button then DC will show the original path.

**Comment by ciros88** [comment]
oh thanks, found it.
it displays original files path and file restoration action, cool. but didn't found the empty trash action.

i need to say this button isn't so..easy to spot. i mean, in my opinion it should be a well visible dedicated button (like in dolphin or krusader), with also the empty-full "animation". what do you think about?

**Comment by alexx2000** [comment]
I don't plan to add this button somewhere else. Maybe I will add "empty trash" command.

**Comment by ciros88** [comment]
the "empty trash" command would be really appreciated

**Comment by timur-g** [comment]
I only now tried this in DC 1.1.32 gamma, build: 2026/01/07, x86_64-Linux-gtk2, Linux Mint 21.3 Virginia, X-Cinnamon (x11)
And I only see Name, Ext, Original path. Date column is mising, to sort by recently deleted. 
I see Date column  in Windows DC.
